### DISCUSSION TITLE
TrustWalletCore/WalletCore pod in Kotlin Multiplatform - almost no classes

### DISCUSSION THREAD

**Discussion by igemonster** [trustwallet/wallet-core#2494] [Ideas & Feature request]
**Describe the bug**
Only a **Crypto** class is available from **cocoapods.WalletCore** in Kotlin  (and CryptoMeta which doesn't look too different).

**To Reproduce**
In a _Kotlin Multiplatform Mobile_ project for iOS and Android, try to access TrustWalletCore cocoapod from Kotlin:
```
// build.gradle.kts (:shared)
cocoapods {
   version = "1.0"
   podfile = project.file("../iosApp/Podfile")
   pod("WalletCore")
}
```
the Podfile is
```
target 'iosApp' do
  pod 'TrustWalletCore'
end
```
This successfully enables import `cocoapods.WalletCore.*` in **shared/iosMain** (see screenshot). Without the above `cocoapods {...}` the import is unavailable. So far so good.

**Expected behavior**
Many classes available for import in Kotlin: WalletCore classes like `HDWallet`, `EthereumSigningInput`, `AnySigner` etc. 

**Screenshots**
![image](https://user-images.githubusercontent.com/3066560/184134123-0a01f30f-d495-490e-80d1-c6e8c317093a.png)

**Additional context**
I can see/use these library classes in Swift, in Xcode via `pod TrustWalletCore` (and `import WalletCore`).

By the looks of it, the available **Crypto** class is _generated_ from the Pod/library [by commonizer](https://fossies.org/linux/kotlin/native/commonizer/README.md) in `0_WalletCore.knm` (about 15 `expect` functions in total - a couple here for illustration):
```
@kotlin.commonizer.ObjCCallable public open external expect fun base58Encode(data: platform.Foundation.NSData): kotlin.String { /* compiled code */ }

@kotlin.commonizer.ObjCCallable public open external expect fun generateMnemonicFromSeed(seed: platform.Foundation.NSData): kotlin.String { /* compiled code */ }
```
**Crypto** has mnemonic-related functionality, as well as `signHash/verifySignature` but not much else.

**Comment by hewigovens** [comment]
`AnySigner` / `HDWallet` might need to add `@objc` to be exposed to Objc then Kotlin, `EthereumSigningInput` is a typealias of a `SwiftProtobuf` Message, I guess the same reason

Inspired from: https://github.com/KodeinKoders/playground-SwiftLib-in-KMMLib#points-of-interest

**Comment by martin-braun** [comment]
Was this addressed or does the team ignore? No offense here, I'm just curious if they are open to make contributing by the community welcoming.
